File: apps/sdkwork-knowledgebase-mini-program/packages/sdkwork-knowledgebase-mp-shell/src/auth/authGate.ts
/**
 * AuthGate integration for the mini program shell.
 *
 * Route guards are shell/runtime responsibilities; capability packages declare
 * auth mode and permission hints only
 * (`APP_CLIENT_ARCHITECTURE_ALIGNMENT_SPEC.md` section 7).
 */
export interface KnowledgebaseMpAuthGateDecision {
  readonly allowed: boolean;
  readonly redirectPagePath?: string;
  readonly reason?: string;
}

export function evaluateKnowledgebaseMpAuthGate(
  auth: "public" | "required",
  isAuthenticated: boolean,
  loginPagePath = "pages/home/index",
): KnowledgebaseMpAuthGateDecision {
  if (auth === "public" || isAuthenticated) {
    return { allowed: true };
  }
  return { allowed: false, redirectPagePath: loginPagePath, reason: "authentication-required" };
}

File: apps/sdkwork-knowledgebase-flutter-mobile/packages/sdkwork_knowledgebase_flutter_mobile_shell/lib/src/auth/auth_gate.dart
/// AuthGate integration for the Flutter mobile shell.
///
/// Route guards are shell/runtime responsibilities. Capability packages
/// declare auth mode and permission hints only.
class SdkworkKnowledgebaseAuthGateDecision {
  const SdkworkKnowledgebaseAuthGateDecision({required this.allowed, this.reason});

  final bool allowed;
  final String? reason;
}

SdkworkKnowledgebaseAuthGateDecision evaluateSdkworkKnowledgebaseAuthGate({
  required bool authRequired,
  required bool isAuthenticated,
}) {
  if (!authRequired || isAuthenticated) {
    return const SdkworkKnowledgebaseAuthGateDecision(allowed: true);
  }
  return const SdkworkKnowledgebaseAuthGateDecision(
    allowed: false,
    reason: 'authentication-required',
  );
}
